tuf/builder.go

```diff
@@ -0,0 +1,402 @@
+package tuf
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/certs"
+	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+// ErrBuildDone is returned when any functions are called on RepoBuilder, and it
+// is already finished building
+type ErrBuildDone struct{}
```

Comment:
non-blocking: seems like this could just be a type instead of an error -- similar to `StopWalk` in `tuf.go`?  It's less clear whether a generic type makes sense here because we have more error cases, but I'm also not sure if the "done building" state should be messaged as an error either.